app/models/account_user.rb

```diff
@@ -33,7 +33,7 @@ class AccountUser < ApplicationRecord
   accepts_nested_attributes_for :account
 
   after_create_commit :notify_creation, :create_notification_setting
-  after_destroy :notify_deletion, :remove_user_from_account
+  after_destroy :notify_deletion, :remove_user_from_account, :unassign_conversations
```

Comment:
let's move this logic to `::Agents::DestroyService` so that all actions that happen on agent delete remain in one place.